Sources/DistributedSystem/DiscoveryManager.swift

```diff
@@ -16,7 +16,7 @@ import PackageConcurrencyHelpers
 final class DiscoveryManager {
     private final class ProcessInfo {
         var channel: (UInt32, Channel)?
-        var pendingServices = [(NodeService, DistributedSystem.ConnectionHandler)]()
+        var pendingServices = [(UUID, NodeService, DistributedSystem.ConnectionHandler)]()
```

Comment:
Maybe it's time to introduce a `struct`?

Reply:
Well, I tried that but at current state it gives more code...